compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -4185,7 +4185,7 @@ declare_lint! {
     Warn,
     "never type fallback affecting unsafe function calls",
     @future_incompatible = FutureIncompatibleInfo {
-        reason: FutureIncompatibilityReason::FutureReleaseSemanticsChange,
+        reason: FutureIncompatibilityReason::EditionAndFutureReleaseError(Edition::Edition2024),
```

Comment:
hm, do we want to maintain the fact that this is a semantics change? i think we kinda care

Reply:
whoops i forgot to finish my sentece -- we kinda care about this difference, right?

Reply:
Yes, we should maintain this fact. Oops ^^'

Reply:
Fixed.